js/client/modules/@arangodb/process-utils.js

```diff
@@ -608,10 +618,16 @@ function executeAndWait (cmd, args, options, valgrindTest, rootDir, circumventCo
     res = executeExternal(cmd, args);
     instanceInfo.pid = res.pid;
     instanceInfo.exitStatus = res;
-    runProcdump(options, instanceInfo, rootDir, res.pid);
-    Object.assign(instanceInfo.exitStatus, 
-                  statusExternal(res.pid, true));
-    stopProcdump(options, instanceInfo);
+    if (runProcdump(options, instanceInfo, rootDir, res.pid)) {
+      Object.assign(instanceInfo.exitStatus, 
+                    statusExternal(res.pid, true));
+      stopProcdump(options, instanceInfo);
```

Comment:
This is executeAndWait function, but we use executeExternal -> runProcdump -> stopProcdump. Why?

Reply:
because of else we couldn't attach procdump - https://github.com/arangodb/arangodb/blob/devel/lib/V8/v8-utils.cpp#L4407 basically does the same in native code. `statusExternal == TRI_CheckExternalProcess`